File: scala/scala-basics/src/test/scala/com/meridal/examples/scalabasics/ExampleCollectionsSpec.scala
package com.meridal.examples.scalabasics

import org.scalatest.funspec.AnyFunSpec
import org.scalatestplus.junit.JUnitRunner
import scala.collection.mutable.Stack
import org.junit.runner.RunWith

@RunWith(classOf[JUnitRunner])
class ExampleCollectionsSpec extends AnyFunSpec {

  val exampleCollections = new ExampleCollections()
  
  describe("An example of collections") {

    it ("should be created with a list with four integers") {
      val list: List[Int] = exampleCollections.list
      assert(list.length == 4)
    }

    it ("should be created with a map with two entries") {
      val map: Map[String, String] = exampleCollections.map
      assert(map.size == 2)
    }
  }
}

File: kotlin/kotlin-basics/src/test/kotlin/com/meridal/examples/kotlinbasics/ExampleCollectionsSpec.kt
package com.meridal.examples.kotlinbasics

import org.spekframework.spek2.Spek
import org.spekframework.spek2.style.specification.describe

class ExampleCollectionsSpec: Spek({

    val exampleCollections = ExampleCollections()

    describe("An example of collections") {

        it ("should be created with a list with four integers") {
            val list = exampleCollections.list
            assert(list.size == 4)
        }

        it ("should be created with a map with two entries") {
            val map = exampleCollections.map
            assert(map.size == 2)
        }

        it ("should be created with a set with four distinct entries") {
            val distinct = exampleCollections.distinct
            assert(distinct.size == 4)
        }
    }
})